### DISCUSSION TITLE
Please help with how to change output save to directory

### DISCUSSION THREAD

**Discussion by TeaseTest** [hlky/stable-diffusion-webui#1173] [Q&A] [answered]
I would like to output all images to one directory, I find it annoying to hunt through 10 different folders to find the output of whichever function I happened to have been using. Ideally I would like to still have the option to toggle each prompt going to a new folder but all in the same root directory.

I'm not great with py syntax, I'm learning but I'm struggling with this one.

For instance, I want all image outputs to be saved in "C:\Users\user\stable-diffusion-webui\outputs\".
If Sort Samples by prompt is active then outputs would be saved in "C:\Users\user\stable-diffusion-webui\outputs\PromptValue\".

I think it's possible with a command line option but I don't know how to implement the change. Does it require editing the webui.py script or the relauncher.py script in additional arguments? If so, where in the code do I make the edit to enter the new directory?

This is from https://github.com/sd-webui/stable-diffusion-webui/wiki/Command-line-options
  ```
--outdir [OUTDIR]     dir to write results to
  --outdir_txt2img [OUTDIR_TXT2IMG]
                        dir to write txt2img results to (overrides --outdir)
  --outdir_img2img [OUTDIR_IMG2IMG]
                        dir to write img2img results to (overrides --outdir)
```

From webui.py file
```
parser.add_argument("--outdir_img2img", type=str, nargs="?", help="dir to write img2img results to (overrides --outdir)", default=None)
parser.add_argument("--outdir_imglab", type=str, nargs="?", help="dir to write imglab results to (overrides --outdir)", default=None)
parser.add_argument("--outdir_txt2img", type=str, nargs="?", help="dir to write txt2img results to (overrides --outdir)", default=None)
parser.add_argument("--outdir", type=str, nargs="?", help="dir to write results to", default=None)
```

Thank you all so much for your help! I love this tool and it's helping me learn so much!

**Comment by outlookhazy** [comment]
I haven't used it, but "Advanced -> Sort samples by prompt" may be what you're looking for

**Reply by TeaseTest** [reply]
Thanks for your reply, that's not what I'm looking for. That option will still put generated images from txt2img, img2img, and the image lab in three different directories.

**Comment by thedyze** [comment, marked as answer]
In webui.py search for: `outpath = opt.outdir_img2img`
Replace that entire line with  `outpath = "../path/toyour/images"`


Also if you do not want extra subfolder called "samples" search for: `sample_path = os.path.join(outpath, "samples")` 
Replace with `sample_path = outpath`

**Reply by TeaseTest** [reply]
Thank you so much thedyze!!!

**Comment by TeaseTest** [comment]
Thank you so much thedyze!!! Your instructions worked perfectly and I was able to extend it to 99.9% of all image saving.

I removed the samples folder creation by changing:

sample_path = os.path.join(outpath, "samples")
to
sample_path = outpath

I was also able to redirect all images and grids to one location by changing all 3 lines below:

outpath = opt.outdir_txt2img or opt.outdir or "outputs/txt2img-samples"
outpath = opt.outdir_img2img or opt.outdir or "outputs/img2img-samples"
outpath = opt.outdir_imglab or opt.outdir or "outputs/imglab-samples"
to
outpath = "C:/Users/username/stable-diffusion-webui/outputs"

For the upcalers and other functions:

outpathDir = os.path.join(outpath,'RealESRGAN')
outpathDir = os.path.join(outpath,'GoBig')
outpathDir = os.path.join(outpath,'LDSR')
outpathDir = os.path.join(outpath,'GoLatent')
outpathDir = os.path.join(outpath,'GFPGAN')
to
outpathDir = outpath

**Reply by thedyze** [reply]
glad it helped :). 
another tip is to have the output folder in a folder such as "c:\out"
because of the max file path length in windows 10, you will retain more of the prompt name in the folder name if you save the files in a folder with a short pathname.
